Detectors/TPC/simulation/include/TPCSimulation/DigitPad.h

```diff
@@ -85,12 +97,16 @@ DigitPad::DigitPad(int pad)
 inline 
 void DigitPad::setDigit(size_t trackID, float charge)
 {
+  static FairRootManager *mgr = FairRootManager::Instance();
+  const int eventID = mgr->GetEntryNr();
+#ifdef TPC_DIGIT_USEFAIRLINKS
+  mMCLinks.AddLink(FairLink(-1, eventID, "MCTrack", trackID));
+#else
   /// the MC ID is encoded such that we can have 999,999 tracks
   /// numbers larger than 1000000 correspond to the event ID
   /// i.e. 12000010 corresponds to event 12 with track ID 10
   /// \todo Faster would be a bit shift
-#ifdef TPC_DIGIT_USEFAIRLINKS
-  mMCLinks.AddLink(FairLink(-1, FairRootManager::Instance()->GetEntryNr(), "MCTrack", trackID));
+  ++mMCID[(eventID)*1000000 + trackID];
```

Comment:
What you say is correct and it has been discussed in similar form among the TPC developers already. However, this commit merely "reverts" to what was already available before ... just to debug and compare to the FairLinks. I will add your comments as future ideas.